report.go

```diff
@@ -1,17 +1,53 @@
 package main
 
-import "fmt"
+import (
+	"fmt"
+	"math"
+	"time"
+)
 
 type Report struct {
-	Fail    uint64
-	Success uint64
+	Fail      uint64
+	Success   uint64
+	Slowest   time.Duration
+	Fastest   time.Duration
+	Total     time.Duration
+	Average   time.Duration
+	ReqPerSec float64
+}
```

Comment:
This `Report` struct is a bit ambiguous in the sense that it's represents the report of a single Get request as well as the cumulative report, and finally it also allows to compute the final report.

I have the feeling that we could reach the same goal if workers were sending a `struct` made of 2 fields only: the request time and the HTTP exit code.

Since the results are merged into a single channel once workers have finished their job, finalization could take care of computing the average and identifying the slowest and fastest requests. As for the total time

What do you think?